Build a basic generic UI

akshatnitd opened the pull request.

ghoshbishakh commented:
![screenshot from 2017-01-27 21-46-22](https://cloud.githubusercontent.com/assets/5156787/22378047/1c8fce60-e4da-11e6-83aa-e3c326d394bf.png)

ghoshbishakh commented on festflow/static/css/main.css:
```diff
@@ -0,0 +1,65 @@
+
+ .jumbotron {
+ 	margin-top: 100px;
+ }
```
Add newline

ghoshbishakh commented on festflow/static/css/main.css:
```diff
@@ -0,0 +1,65 @@
+	color: #4285f4;
+}
+
+body {
```
are you sure you want to put a background image in body?

ghoshbishakh commented on festflow/static/css/main.css:
```diff
@@ -0,0 +1,65 @@
+.navbar {
+	background-color: #4285f4;
```
Add proper comments to separate different css for different pages or even better use different css files for different views. And a common one.

akshatnitd commented:
Okay sure, Will make the necessary changes.

ghoshbishakh commented on festflow/templates/festflow/about.html:
```diff
@@ -1,13 +1,26 @@
 <html>
 <head>
     <title>ABOUT PAGE</title>
+    <link rel="stylesheet" type="text/css" href='{% static "css/main.css" %}'/>
```
first put meta tags. then link tags

ghoshbishakh commented on festflow/templates/festflow/about.html:
```diff
@@ -1,13 +1,26 @@
+    <link rel="stylesheet" type="text/css" href='{% static "css/main.css" %}'/>
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+  	<link rel="stylesheet" type="text/css" href='{% static "css/bootstrap.min.css" %}'/>
+  	<link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
```
Please dwnload any font you want to use to static

ghoshbishakh commented on festflow/templates/festflow/about.html:
```diff
@@ -1,13 +1,26 @@
+		    	{% if content %}
+			        {{ content | safe }}
+			    {% endif %}
+			</div>
```
Use proper indentation

ghoshbishakh commented on festflow/templates/festflow/about.html:
```diff
@@ -1,13 +1,26 @@
+  		</div>
+	</div>
 </body>
 </html>
```
newline

ghoshbishakh commented on festflow/templates/festflow/index.html:
```diff
@@ -4,12 +4,23 @@
   <title>Home Page</title>
+  <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1" />
   <link rel="stylesheet" type="text/css" href='{% static "css/bootstrap.min.css" %}'/>
```
Use a comment to specify what stuff you are importing

ghoshbishakh commented on festflow/templates/festflow/index.html:
```diff
@@ -4,12 +4,23 @@
+  <div class="title">
+  	<div class="logo">
+  		<a href="#home">
+  			<img src="{% static "images/logo.png" %}" alt="Logo Brand">
```
use static font

ghoshbishakh commented on festflow/templates/festflow/index.html:
```diff
@@ -4,12 +4,23 @@
+  	<h4>A one-liner description of the fest</h4>
+  <div>
 </body>
 <script type="text/javascript" href='{% static "js/jquery.min.js"%}'></script>
```
add comment to specify what stuff you are importing

ghoshbishakh commented on festflow/templates/festflow/navbar.html:
```diff
@@ -1,12 +1,25 @@
-    </ul>
   </div>
-</nav>
+</nav>
```
newline

akshatnitd commented:
![1](https://cloud.githubusercontent.com/assets/12656846/22380602/03f8983c-e4e4-11e6-9afb-bb157fea7c21.jpg)
How does this look?

ghoshbishakh commented:
@akshatnitd looks good

ghoshbishakh commented:
@akshatnitd do not change the PR. Push in this one. Make the changes. Commit and squash.

akshatnitd commented:
Made the changes (Y)

ghoshbishakh commented on festflow/static/css/about.css:
```diff
@@ -0,0 +1,17 @@
+	background-color: #4285f4;
+	height: 5px;
+}
+ 
```
newline

ghoshbishakh commented on festflow/templates/festflow/about.html:
```diff
@@ -1,13 +1,34 @@
 <html>
 <head>
-    <title>ABOUT PAGE</title>
+	<title>ABOUT PAGE</title>
```
indent properly

nkprince007 replied:
Just two spaces please, because, all other markup files use the similar indents.

ghoshbishakh commented on festflow/templates/festflow/about.html:
```diff
@@ -1,13 +1,34 @@
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <!-- LOAD CSS AND BASIC JS -->
+  
```
no reason to indent after comments

ghoshbishakh commented on festflow/templates/festflow/index.html:
```diff
@@ -4,13 +4,31 @@
   <title>Home Page</title>
+  <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <!-- LOAD CSS AND BASIC JS -->
```
use uniform indentation. either 4 spaces or a tab or 2 spaces

akshatnitd commented:
Have made the changes. Let me know if any other changes to be made. Will be working on the other views now.

nkprince007 commented on festflow/static/css/main.css:
```diff
@@ -0,0 +1,49 @@
+	background-color: #4285f4;
+	text-decoration: none;
+	-webkit-box-shadow: 5px 5px 5px #aaa;
+    -moz-box-shadow: 5px 5px 5px #aaa;
```
messed up indentation here.

nkprince007 commented on festflow/templates/festflow/index.html:
```diff
@@ -4,13 +4,31 @@
-  <h1>Home Page</h1>
+  <div class="title">
+    <div class="logo">
+  		<img src="{% static "images/logo.png" %}" alt="Logo Brand">
```
another messed up indentation here.

ghoshbishakh commented:
@akshatnitd In about page there is no navbar. Also images do not have a max width.
![screenshot from 2017-01-28 11-30-54](https://cloud.githubusercontent.com/assets/5156787/22394713/59a8449e-e54d-11e6-98bc-5ead4493d02e.png)
![screenshot from 2017-01-28 11-31-56](https://cloud.githubusercontent.com/assets/5156787/22394716/65348f7a-e54d-11e6-9d86-dca97c283bc9.png)

akshatnitd commented:
@nkprince007 @ghoshbishakh Had some issues with the text editor. Fixed the indent issues. Added navbar to the about page and added max-width to the images.

ghoshbishakh commented:
@akshatnitd you have not made any changes. Please push

akshatnitd commented:
@ghoshbishakh Doing that. Should I also set any max-height for the image inside the about container?

ghoshbishakh commented:
@akshatnitd no maxheight is meaningless. Just use `max-width: 100%`

akshatnitd commented:
@ghoshbishakh Done the changes.

ghoshbishakh commented:
![screenshot from 2017-01-28 15-02-23](https://cloud.githubusercontent.com/assets/5156787/22395665/c962424a-e56a-11e6-9f85-445ddfc7bc6b.png)


@akshatnitd too much margin top in the about page. Reduce the gap between nav and the content.

And one bug: In mobile view the nav links disappear and no dropdown appears.

ghoshbishakh commented on festflow/templates/festflow/about.html:
```diff
@@ -1,13 +1,34 @@
-        ABOUT PAGE
-    {% endif %}
+  <div class="container">
+    <div class="jumbotron">
```
Do not use jumbotron. Use row and a column

akshatnitd commented:
Will resolve these issues :)

ghoshbishakh commented on festflow/templates/festflow/event_view.html:
```diff
@@ -1,29 +1,67 @@
+			<div class="subdash"></div>
+			<br />
+			<br />
+			<p>{{ event.description }}</p>
```
use {{ event.description|safe }}

akshatnitd replied:
Will make the change.

ghoshbishakh commented:
@akshatnitd Login page has issues. Plase make sure buttons resize properly.
![screenshot from 2017-01-29 12-36-13](https://cloud.githubusercontent.com/assets/5156787/22402508/971287c2-e61f-11e6-9520-3ef84c8721a8.png)

ghoshbishakh commented:
TODO (do it later)
@akshatnitd In the events page try to use a library like http://masonry.desandro.com/

That will fix the spacing issue.
![screenshot from 2017-01-29 12-46-49](https://cloud.githubusercontent.com/assets/5156787/22402547/348ad486-e621-11e6-90cb-3a9f3f0fa317.png)

nkprince007 commented:
Masonry is awesome, I've used it on muZic-electron(one of my repos).. 

@ghoshbishakh @akshatnitd 👍 for masonry grid.

ghoshbishakh commented:
@akshatnitd Looks like you are not using template inheritance. It is really simple. Just go through this post: https://tutorial.djangogirls.org/en/template_extending/

Now create a `base.html` that will contain all css and js imports and navbar and title and meta tags.
Then just extend the body part.

akshatnitd commented:
@ghoshbishakh Will make the changes, and will implement the mansory grid after all the pages are designed :+1:

akshatnitd commented:
@ghoshbishakh Checking the linting issues :+1:

ghoshbishakh commented:
@akshatnitd excellend job with the linting! Keep it up

akshatnitd commented:
@ghoshbishakh Cool :+1: Working on other templates now!

ghoshbishakh commented:
@akshatnitd remember tomorrow is your submisssion deadline
its Noon 12. So you have to submit it by morning

ghoshbishakh commented:
@akshatnitd You have probably specified some larger font for paragraphs in events_view page. As a result texts in lists appear small. Please check.
![screenshot from 2017-01-30 23-24-29](https://cloud.githubusercontent.com/assets/5156787/22434752/59bf9714-e743-11e6-9e92-6de79919c126.png)

akshatnitd commented:
@ghoshbishakh Made the change, not all tags under jumbotron are of same size !
Also pushed the code for other templates.

ghoshbishakh commented:
@akshatnitd please change the last commit message. Be specific about what remaing templates

ghoshbishakh commented:
@akshatnitd atleast remove the word "remaining".

ghoshbishakh commented on festflow/templates/404.html:
```diff
@@ -0,0 +1,16 @@
+            <img src="{% static "images/logo.png" %}" class="img-responsive" alt="Logo Brand">
+        </div>
+        <h1>404 ERROR</h1>
+        <h4>This page isn't available. Sorry about that.</h4>
```
@akshatnitd  Please change this text to something like 'The page you are looking for does not exist.'

akshatnitd replied:
Okay sure :)

ghoshbishakh commented on festflow/templates/500.html:
```diff
@@ -0,0 +1,16 @@
+            <img src="{% static "images/logo.png" %}" class="img-responsive" alt="Logo Brand">
+        </div>
+        <h1>500 ERROR</h1>
+        <h4>This page isn't available. Sorry about that.</h4>
```
@akshatnitd please change the text to something like.. 'Opps something went wrong.. Internal Server Error'

akshatnitd replied:
Okay sure :)

ghoshbishakh commented:
@akshatnitd Submission LINK: 
https://docs.google.com/forms/d/e/1FAIpQLScC9f0OZVQUM2pfZZ7MLzs4Z1JvfzK25oKDXChhEC1yrWVPVQ/viewform?c=0&w=1

Please share the gist with me before submitting.

akshatnitd commented:
@ghoshbishakh Thanks for the submission link! Will share before submitting.

akshatnitd commented:
Made the final commit for error message change on 404.html and 500.html.

And glad to see the green ticks :smile: :smiley:

ghoshbishakh commented:
@akshatnitd I see you have included masonry but not used it? You can do it in a separate PR.
@nkprince007 please check this and confirm merge!

akshatnitd commented:
@ghoshbishakh I was just trying to use it.
Will do it in separate PR.
Should I remove the mansory static files?

ghoshbishakh commented:
@akshatnitd no let it be. Let this pr be merged first

akshatnitd commented:
Okay.. Will do that part later.

ᐧ

On Wed, Feb 1, 2017 at 11:35 AM, Bishakh Ghosh <notifications@github.com>
wrote:

> @akshatnitd <https://github.com/akshatnitd> no let it be. Let this pr be
> merged first
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NIT-dgp/festcms/pull/92#issuecomment-276579904>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AMEgzlzGovdaTwwqsjloWyousdolg0Zfks5rYCCQgaJpZM4Lv7Zk>
> .
>

nkprince007 commented:
Looks good to me @ghoshbishakh
